docs/ide/reference/options-text-editor-csharp-advanced.md

```diff
@@ -26,6 +26,70 @@ Use the **Advanced** options page to modify the settings for editor formatting,
 
    Enables code analysis on all files in the solution, not just open code files. For more information, see [Full solution analysis](../../code-quality/how-to-enable-and-disable-full-solution-analysis-for-managed-code.md).
 
+## Using Directives
+
+- Place 'System' directives first when sorting usings
+
+   When selected, when you select **Remove and Sort Usings**, Visual Studio sorts the `using` directives and places the 'System' namespaces at the top of the list.
+
+   Before sorting:
+
+   ```csharp
+   using AutoMapper;
+   using FluentValidation;
+   using System.Collections.Generic;
+   using System.Linq;
+   using Newtonsoft.Json;
+   using System;
+   ```
+   
+   After sorting:
+
+   ```csharp
+   using System;
+   using System.Collections.Generic;
+   using System.Linq;
+   using AutoMapper;
+   using FluentValidation;
+   using Newtonsoft.Json;
+   ```
+   
+- Separate using directive groups
+
+   When selected, when you select **Remove and Sort Usings**, Visual Studio separates `using` directives by inserting an empty line between groups of directives.
+
+   Before sorting:
+
+   ```csharp
+   using AutoMapper;
+   using FluentValidation;
+   using System.Collections.Generic;
+   using System.Linq;
+   using Newtonsoft.Json;
+   using System;
+   ```
+   
+   After sorting:
+   
+   ```csharp
+   using AutoMapper;
+   
+   using FluentValidation;
+   
+   using Newtonsoft.Json;
+   
+   using System;
+   using System.Collections.Generic;
+   using System.Linq;
+   ```
+   
+- Add usings for types in reference assemblies and NuGet packages 
```

Comment:
From the name of this option, I'm not sure that the description that VS installs the NuGet package is correct. I think it's more likely that this option means that VS will suggest using directives to add for types you're using that are in referenced assemblies or NuGet packages - not that it will install the NuGet package.

Reply:
Hi, that's exactly what it does. 

If you have the option selected it will look for and install types from un-referenced and installed nuget packages.

I created a new blank console app and if add the code var context = new DbContext, the lightbulb will allow me to install the latest version of the nuget package, or use the package manager to install a specific version.

as you can see in the screenshot below, i dont have any nuget pkgs installed but it is now allowing me install the EntityFramework pkg.

![image](https://user-images.githubusercontent.com/16678850/46836700-5a968000-cdfe-11e8-9356-a8c21736d177.png)

Reply:
@dixonsj is correct! We added the option in 2015.3 ([release notes](https://docs.microsoft.com/en-us/visualstudio/releasenotes/vs2015-update3-vs#csharpandvb)) though it's never been turned on by default! Very useful and we definitely need to document this better. The only other place it's mentioned that I could find is one of Scott Hansleman's[ blog posts.](https://www.hanselman.com/blog/VisualStudio2017CanAutomaticallyRecommendNuGetPackagesForUnknownTypes.aspx)